src/installer.ts

```diff
@@ -77,20 +80,57 @@ async function acquireGo(version: string): Promise<string> {
     throw new Error('Temp directory not set');
   }
 
-  if (osPlat == 'win32') {
+  // gotip is always .tar.gz.
+  if (osPlat == 'win32' && !gotip) {
     extPath = await tc.extractZip(downloadPath);
   } else {
     extPath = await tc.extractTar(downloadPath);
   }
 
+  let toolRoot = extPath;
+  if (!gotip) {
+    toolRoot = path.join(toolRoot, 'go');
+    version = normalizeVersion(version);
+  }
+
+  //
+  // Build gotip from source.
+  //
+  if (gotip) {
+    // Version embed in the executable. Required for build.
+    let versionFile = path.join(toolRoot, 'VERSION');
+    fs.writeFileSync(versionFile, `gotip-${version}`);
+
+    const options = {
+      cwd: path.join(toolRoot, 'src'),
+      env: {
+        ...process.env,
+        // Embed in the executable, set it explicitly for reproducible builds.
+        'GOROOT_FINAL': '/go',
+      },
+    };
+    // TODO: is it OK to run make.bash on Windows?
+    // Anyway, I have no idea how to run make.bat using exec. Help would be appreciated.
+    let rc: number = await exec.exec('bash', ['make.bash'], options);
```

Comment:
Not okay to run `make.bash` on windows. Need to run `make.bat`. https://golang.org/doc/install/source#install